### PR TITLE
Go to axios

### PR DESCRIPTION
**Pull request by isabloCh**

### CODE REVIEW

File: app/controllers/api/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Api
+  class BooksController < ActionController::Base
+
+    def index
+      @books = Book.all
```
Comment (resivalex) [inline app/controllers/api/books_controller.rb]: использовать `books` вместо `@books`

File: app/views/books/index.html.erb [hunk 2]
```diff
@@ -1 +1 @@
-<div id="books_list" data="<%= @books.to_json %>" />
+<div id="books_list" data="<%= @books.to_json %>" />
```
Comment (resivalex) [inline app/views/books/index.html.erb]: перенос строки в конце файла

Comment (resivalex) [reply app/views/books/index.html.erb]: атрибут `data` больше не нужен

File: config/routes.rb [hunk 3]
```diff
@@ -2,6 +2,10 @@
   devise_for :users, ActiveAdmin::Devise.config
   ActiveAdmin.routes(self)
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  root to: 'books#index'
+  root to:  'books#index'
```
Comment (resivalex) [inline config/routes.rb]: лишний пробел

**Review by resivalex — CHANGES_REQUESTED** [review]
👍

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -1,11 +1,5 @@
 class BooksController < ApplicationController
```
Comment (Loremaster) [inline app/controllers/books_controller.rb:1]: Looks like this controller can be deleted now

File: app/javascript/vue/App.vue [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
+      }
+    },
     components: {
       BookShelf
     },
     created() {
-      this.books = bookData.books
+      booksData.then((books) => {
+        this.books = books
+      })
```
Comment (Loremaster) [inline app/javascript/vue/App.vue]: What will you do if there will be an error while fetching data through ajax? You should let your users know about that.

Comment (resivalex) [reply app/javascript/vue/App.vue]: 👍

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (isabloCh) [reply app/controllers/books_controller.rb:1]: No, we need it. What we will have root without it?

File: app/javascript/vue/App.vue (hunk 5 above)
Comment (isabloCh) [reply app/javascript/vue/App.vue]: Perhaps it makes sense to do this not in the controller of the rails, but in the vue?
```
axios.get('/user', {
  })
  .then(function (response) {
  })
  .catch(function (error) {
  });
```

Comment (Loremaster) [reply app/javascript/vue/App.vue]: @isabloCh yes you handle this type of errors in vue

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (Loremaster) [reply app/controllers/books_controller.rb:1]: Where do you need it?

File: app/javascript/vue/App.vue (hunk 5 above)
Comment (resivalex) [reply app/javascript/vue/App.vue]:
```
booksData.then((books) => {
  this.books = books
}, (errorMessage) => {
  alert(errorMessage) // or something else
})
```

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (isabloCh) [reply app/controllers/books_controller.rb:1]: routes.rb file

File: app/javascript/vue/App.vue (hunk 5 above)
Comment (Loremaster) [reply app/javascript/vue/App.vue]: Good idea @resivalex!

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (Loremaster) [reply app/controllers/books_controller.rb:1]: Sorry I mean why do you need this controller? Do you render vue component in `BooksController#index`?

Comment (resivalex) [reply app/controllers/books_controller.rb:1]: I suppose, that @isabloCh gonna create individual Vue components for every page

@Loremaster, Do you mean a single component to rule them all?

Comment (isabloCh) [reply app/controllers/books_controller.rb:1]: yes

Comment (Loremaster) [reply app/controllers/books_controller.rb:1]: @resivalex yep I think it’ll be good to have it. We don’t have to do it in this PR

File: app/javascript/vue/books-data.js [hunk 6]
```diff
@@ -1,9 +1,5 @@
...
-  let books = JSON.parse(el.getAttribute('data'));
+import axios from 'axios'
 
-  console.log(books)
-  return {
-    books
-  }
-}
+export default new Promise((resolve) => {
+  axios('/api/books').then((response) => resolve(response.data) )
```
Comment (resivalex) [inline app/javascript/vue/books-data.js]: лишний пробел перед последней скобкой

**Review by resivalex — CHANGES_REQUESTED** [review]

File: app/javascript/vue/books-data.js [hunk 7]
```diff
@@ -1,9 +1,10 @@
...
-  console.log(books)
-  return {
-    books
-  }
-}
+export default new Promise((resolve) => {
+  axios('/api/books')
+    .then((response) => resolve(response.data))
+    .catch(function (error) {
+      alert('ошибонька вышла');
```
Comment (resivalex) [inline app/javascript/vue/books-data.js]: `Promise` should trigger `reject` and pass error message to it

**Review by resivalex — CHANGES_REQUESTED** [review]

File: app/javascript/vue/books-data.js [hunk 8]
```diff
@@ -1,9 +1,10 @@
...
 
-  console.log(books)
-  return {
-    books
-  }
-}
+export default new Promise((resolve) => {
+  axios('/api/books')
+    .then((response) => resolve(response.data))
+    .catch(function (error) {
```
Comment (resivalex) [inline app/javascript/vue/books-data.js]: Use shorthand for anonymous function

**Review by resivalex — CHANGES_REQUESTED** [review]

**Comment by isabloCh** [comment]
@Loremaster  @resivalex   need to disable verification  
 image
the vue swears
image

**Comment by resivalex** [comment]
`return new ...`

**Review by resivalex — APPROVED** [review]
❤️, lovely...

**Review by Loremaster — APPROVED** [review]

**Pull request merged**